chain/chain/src/runtime/mod.rs

```diff
@@ -748,66 +748,59 @@ impl RuntimeAdapter for NightshadeRuntime {
                     continue;
                 }
 
                 // Verifying the transaction is on the same chain and hasn't expired yet.
-                if !chain_validate(&signed_tx) {
-                    tracing::trace!(target: "runtime", tx=?signed_tx.get_hash(), "discarding transaction that failed chain validation");
+                if !chain_validate(&validated_tx.to_signed_tx()) {
+                    tracing::trace!(target: "runtime", tx=?validated_tx.get_hash(), "discarding transaction that failed chain validation");
                     rejected_invalid_for_chain += 1;
                     continue;
                 }
 
-                let (verify_result, signed_tx) =
-                    match validate_transaction(runtime_config, signed_tx, protocol_version) {
-                        Err((err, signed_tx)) => (Err(err), signed_tx),
-                        Ok(validated_tx) => {
-                            let res = tx_cost(
-                                runtime_config,
-                                &validated_tx,
-                                prev_block.next_gas_price,
-                                protocol_version,
-                            )
-                            .map_err(InvalidTxError::from)
-                            .and_then(|cost| {
-                                verify_and_charge_tx_ephemeral(
-                                    runtime_config,
-                                    &state_update,
-                                    &validated_tx,
-                                    &cost,
-                                    Some(next_block_height),
-                                    protocol_version,
-                                )
-                            })
-                            .and_then(|vr| {
-                                commit_charging_for_tx(
-                                    &mut state_update,
-                                    &validated_tx,
-                                    &vr.signer,
-                                    &vr.access_key,
-                                );
-                                Ok(vr)
-                            });
-                            (res, validated_tx.into_signed_tx())
-                        }
-                    };
+                let verify_result = tx_cost(
+                    runtime_config,
+                    &validated_tx,
+                    prev_block.next_gas_price,
+                    protocol_version,
+                )
+                .map_err(InvalidTxError::from)
+                .and_then(|cost| {
+                    verify_and_charge_tx_ephemeral(
+                        runtime_config,
+                        &state_update,
+                        &validated_tx,
+                        &cost,
+                        Some(next_block_height),
+                        protocol_version,
+                    )
+                })
+                .and_then(|vr| {
```

Comment:
mini nit: can you rename `vr` to what it stands for?

Reply:
done.